source/common/http/conn_manager_impl.cc

```diff
@@ -1766,6 +1766,11 @@ void ConnectionManagerImpl::ActiveStream::recreateStream(
   // Make sure to not check for deferred close as we'll be immediately creating a new stream.
   connection_manager_.doEndStream(*this, /*check_for_deferred_close*/ false);
 
+  auto oldAccount = response_encoder->getStream().account();
+  if (oldAccount != nullptr) {
+    oldAccount->clearDownstream();
+  }
+
```

Comment:
The slices in the `request_data` still are referring to the old account, we should have them transfered to refer to the account of the recreated stream (crediting the old account which will be destroyed at the end of this function)

Reply:
Probably, we can re-use the old account because the downstream connection is still the same.
WDYT?

Reply:
yeah I think that'd work and be simpler. We would need to modify `ConnectionManagerImpl::newStream` for the case there's an existing account associated which there will be if we're recreating the stream.